ray-operator/controllers/ray/common/service.go

```diff
@@ -118,6 +118,12 @@ func BuildServiceForHeadPod(ctx context.Context, cluster rayv1.RayCluster, label
 			Type:     defaultType,
 		},
 	}
+	if defaultType == "" || defaultType == corev1.ServiceTypeClusterIP {
+		// Use headless service for the Head Pod by default,
+		// since there is no sense for Head Pods to share a same virtual IP.
+		headService.Spec.ClusterIP = corev1.ClusterIPNone
```

Comment:
Could this be a breaking change? Any RayCluster today that is accessed by a ClusterIP would lose connectivity when upgrading to a kuberay version that now creates headless services.

Reply:
I understand that this change does not recreate the head Service, but any user that creates RayCluster and automatically uses the generated ClusterIP could be impacted.

Reply:
Yes, they will be broken if they don't support headless service in their code.

Reply:
Most service mesh and ingress solutions support headless services, but still, depend on how users configure them. They may still be broken.

Reply:
> How common is the hairpinning issue?

Hairpinning is uncommon. Only companies with very high security standards typically disable it.

> but any user that creates RayCluster and automatically uses the generated ClusterIP could be impacted.

Is it common for users to use a ClusterIP instead of the DNS name of a Kubernetes service?

Reply:
Some users may query the ClusterIP through k8s API directly, just like what we did in the `getHeadServiceIPAndName` function, for whatever reasons. But I believe this is also uncommon.